src/main/java/org/mule/extension/email/internal/mailbox/imap/IMAPPollingSource.java

```diff
@@ -87,4 +115,32 @@ public void onTerminate() {
     endUsingFolder();
   }
 
+  @Override
+  public void doStart() throws ConnectionException {
+    super.doStart();
+    if (this.getPredicateBuilder().isPresent()) {
+      this.getPredicateBuilder().get().initializeRemoteSearchTerm();
+    }
+  }
+
+  @Override
+  protected Message[] getMessages(Folder openFolder) {
+    try {
+      if (!(this.isRemoteSearchFilterEnabled() && getPredicateBuilder().isPresent()
+          && getPredicateBuilder().get().getRemoteSearchTerm().isPresent())) {
+        //Filters will be applied locally.
+        return super.getMessages(openFolder);
+      }
+
+      try {
+        return openFolder.search(getPredicateBuilder().get().getRemoteSearchTerm().get());
+      } catch (MessagingException e) {
+        return openFolder.getMessages();
```

Comment:
Is there any difference betweeen openFolder.getMessages() and super.getMessages(openFolder) ?